continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala

```diff
@@ -72,4 +72,414 @@ class ContinuationsPluginSuite extends FunSuite, CompilerFixtures {
+           |        safeContinuation.getOrThrow()
+           |      }
+           |  }
+           |}
+           |""".stripMargin
+      // format: on
+
+      checkContinuations(source) {
+        case (tree, _) =>
+          assertNoDiff(compileSourceIdentifier.replaceAllIn(tree.show, ""), expected)
+      }
+  }
+
+  compilerContextWithContinuationsPlugin.test(
+    "it should convert simple suspended def with no parameters with a Left input") {
+    implicit givenContext =>
+
+      val source =
+        """|
+           |package continuations
+           |
+           |def foo()(using Suspend): Int =
+           |  Continuation.suspendContinuation[Int] { continuation => continuation.resume(Left(new Exception("error"))) }
+           |""".stripMargin
+
+      // format: off
+      val expected =
+        """|
+           |package continuations {
+           |  final lazy module val compileFromString$package: 
+           |    continuations.compileFromString$package
+           |   = new continuations.compileFromString$package()
+           |  @SourceFile("compileFromString.scala") final module class 
+           |    compileFromString$package
+           |  () extends Object() { this: continuations.compileFromString$package.type =>
+           |    private def writeReplace(): AnyRef = 
+           |      new scala.runtime.ModuleSerializationProxy(classOf[continuations.compileFromString$package.type])
+           |    def foo(completion: continuations.Continuation[Int]): Any | Null | continuations.Continuation.State.Suspended.type = 
+           |      {
+           |        val continuation1: continuations.Continuation[Int] = completion
+           |        val safeContinuation: continuations.SafeContinuation[Int] = 
+           |          new continuations.SafeContinuation[Int](continuations.intrinsics.IntrinsicsJvm$package.intercepted[Int](continuation1)(), 
+           |            continuations.Continuation.State.Undecided
+           |          )
+           |        val suspendContinuation: Int = 0
+           |        safeContinuation.resume(Left.apply[Exception, Nothing](new Exception("error")))
+           |        safeContinuation.getOrThrow()
+           |      }
+           |  }
+           |}
+           |""".stripMargin
+      // format: on
+
+      checkContinuations(source) {
+        case (tree, _) =>
+          assertNoDiff(compileSourceIdentifier.replaceAllIn(tree.show, ""), expected)
+      }
+  }
+
+  compilerContextWithContinuationsPlugin.test(
+    "it should convert simple suspended def with no parameters keeping the rows before the suspend call") {
+    implicit givenContext =>
+
+      val source =
+        """|
+           |package continuations
+           |
+           |def foo()(using Suspend): Int = {
+           |  val x = 5
+           |  println("HI")
+           |  Continuation.suspendContinuation[Int] { continuation => continuation.resume(Right(1)) }
+           |}
```

Comment:
We should probably add this to the coroutines-test to see what it outputs in this case. Can you make a clickup for that and add it as a subtask of the compiler plugin task @gregorath?

Reply:
yes I tested that and it keep the rows. 
I will add a task though because it is actually about what it is being returned eg
```
return 10
Continuation.suspendContinuation[Int] { continuation => continuation.resume(Right(1)) }
```

creates a more complicate tree with `switch`.